polkadot/runtime/parachains/src/runtime_api_impl/vstaging.rs

```diff
@@ -15,3 +15,30 @@
 // along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
 
 //! Put implementations of functions from staging APIs here.
+
+use crate::shared;
+use primitives::ValidatorIndex;
+use sp_std::{collections::btree_map::BTreeMap, prelude::Vec};
+
+/// Implementation for `DisabledValidators`
+// CAVEAT: this should only be called on the node side
+// as it might produce incorrect results on session boundaries
+pub fn disabled_validators<T>() -> Vec<ValidatorIndex>
+where
+	T: pallet_session::Config + shared::Config,
+{
+	let shuffled_indices = <shared::Pallet<T>>::active_validator_indices();
+	// mapping from raw validator index to `ValidatorIndex`
+	// this computation is the same within a session, but should be cheap
+	let reverse_index = shuffled_indices
+		.iter()
+		.enumerate()
+		.map(|(i, v)| (v.0, ValidatorIndex(i as u32)))
+		.collect::<BTreeMap<u32, ValidatorIndex>>();
+
+	// we might have disabled validators who are not parachain validators
+	<pallet_session::Pallet<T>>::disabled_validators()
```

Comment:
I think I get what's going on here, but if you could check my thinking that would be awesome.

With inputs:
`shuffled_indices = [6, 9, 1]`
`disabled_validators = [9, 6]`

We'd get the steps

reverse_index = BTreeMap with entries (6, 1), (9, 2), (1, 3)

returned value = vec![2, 1]

Reply:
Almost, the only correction is that indices are 0-based, not 1-based.